Update extension update test

nasbyj opened the pull request:
Update the minor-version-upgrade (aka ALTER EXTENSION UPDATE) test to
use the new source/from tags instead of hard-coding that to specific version
branches.

Signed-off-by: Jim C. Nasby <nasbyj@amazon.com>


### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

nasbyj committed:
Update tests for 2.1.STABLE

Change the default engine download for tests to be
BABEL_2_1_STABLE__PG_14_3. Also pdate the minor-version-upgrade (aka
ALTER EXTENSION UPDATE) test to use the new source/from tags.

Signed-off-by: Jim C. Nasby <nasbyj@amazon.com>

nasbyj committed:
Update extension update test for new versions

Update the minor-version-upgrade (aka ALTER EXTENSION UPDATE) test to
use the next version engine branch and the new source/from tags.

Signed-off-by: Jim C. Nasby <nasbyj@amazon.com>

nasbyj committed:
Don't filter job by branch

Job wasn't actually running on the stable branch.

Signed-off-by: Jim C. Nasby <nasbyj@amazon.com>

thephantomthief committed:
Use composite actions to build and install older version extensions (#158)

Signed-off-by: Sharu Goel <goelshar@amazon.com>
Co-authored-by: Sharu Goel <goelshar@amazon.com>

2jungkook commented on .github/workflows/minor-version-upgrade.yml:
```diff
@@ -1,19 +1,18 @@
+      # TODO: hard-coding this means that we're not actually testing mVU for
+      # any PR that makes engine changes. This needs to be changed use
+      # something similar to .github/scripts/clone_engine_repo.
       ENGINE_VER_TO: BABEL_2_X_DEV__PG_14_2
```
Target engine version should be `BABEL_2_1_STABLE__PG_14_3`.

nasbyj replied:
ugh, good catch. Really need to get this stuff to use the `clone_engine_repo` script. :(

nasbyj replied:
Fixed

nasbyj committed:
Change mVU script to use 2.1.STABLE

Signed-off-by: Jim C. Nasby <nasbyj@amazon.com>

nasbyj committed:
Fix handing of `clone_engine_repo.conf`

`clone_engine_repo` was mistakenly checking to see if the config file
was executable, instead of whether it's merely readable. Fix that, throw
an error if the file exists but isn't readable, and throw a warning if
the config file doesn't exist.

Signed-off-by: Jim C. Nasby <nasbyj@amazon.com>
